internal/config/db.go

```diff
@@ -70,6 +71,16 @@ type PostgresReadReplica struct {
 	Hostname string `yaml:"hostname" json:"hostname" env:"HOSTNAME"`
 }
 
+// PostgresPool defines configuration for connections pooling.
+type PostgresPool struct {
+	Name              string `yaml:"name" json:"name" env:"NAME" env-default:"pgx"`
+	MaxConns          int32  `yaml:"max_conns" json:"max_conns" env:"MAX_CONNS" env-default:"50"`
+	MinConns          int32  `yaml:"min_conns" json:"min_conns" env:"MIN_CONNS" env-default:"2"`
+	MaxConnLifetime   string `yaml:"max_conn_lifetime" json:"max_conn_lifetime" env:"MAX_CONN_LIFETIME" env-default:"1h"`
+	MaxConnIdleTime   string `yaml:"max_conn_idle_time" json:"max_conn_idle_time" env:"MAX_CONN_IDLE_TIME" env-default:"1h"`    //nolint:lll
+	HealthCheckPeriod string `yaml:"health_check_period" json:"health_check_period" env:"HEALTH_CHECK_PERIOD" env-default:"1m"` //nolint:lll
```

Comment:
This should probably be disabled by default, right?

Reply:
Ohh, completely sidestepped on the fact that the pgx pool implementation implements its own health check too... derp, mind me.